tests/hla/base_test.py

```diff
@@ -0,0 +1,405 @@
+
+            for k in range(0,3):
+                i = indexes[k]
+                diff.append(distances[i][0])
+                fd.write("\n%s distance = %f flux difference = %f\n" % (stats[k], distances[i][0], distances[i][1]))
+
+                for j in range(2, 4):
+                    ylo = int(distances[i][j][0]) - (1+self.buff)
+                    yhi = int(distances[i][j][0]) + (2+self.buff)
+                    xlo = int(distances[i][j][1]) - (1+self.buff)
+                    xhi = int(distances[i][j][1]) + (2+self.buff)
+                    subimage = images[j][ylo:yhi,xlo:xhi]
+                    fd.write("\n%s %s image centroid = (%f,%f) image flux = %f\n" %
+                             (stats[k], im_type[j], distances[i][j][0], distances[i][j][1], distances[i][j][2]))
+                    fd.write(str(subimage) + "\n")
+
+        fd.close()
+        return tuple(diff)
+
+    def make_point_image(self, input_image, point, value):
+        """
+        Create an image with a single point set
+        """
+        output_image = np.zeros(input_image.shape, dtype=input_image.dtype)
+        output_image[point] = value
+        return output_image
+
+    def make_grid_image(self, input_image, spacing, value):
+        """
+        Create an image with points on a grid set
+        """
+        output_image = np.zeros(input_image.shape, dtype=input_image.dtype)
+
+        shape = output_image.shape
+        for y in range(spacing//2, shape[0], spacing):
+            for x in range(spacing//2, shape[1], spacing):
+                output_image[y,x] = value
+
+        return output_image
+
+    def print_wcs(self, title, wcs):
+        """
+        Print the wcs header cards
+        """
+        print("=== %s ===" % title)
+        print(wcs.to_header_string())
+
+
+    def read_image(self, filename):
+        """
+        Read the image from a fits file
+        """
+        hdu = fits.open(filename)
+
+        image = hdu[1].data
+        hdu.close()
+        return image
+
+    def read_wcs(self, filename):
+        """
+        Read the wcs of a fits file
+        """
+        hdu = fits.open(filename)
+
+        wcs = stwcs.wcsutil.HSTWCS(hdu, 1)
+        hdu.close()
+        return wcs
+
+    def write_wcs(self, hdu, image_wcs):
+        """
+        Update header with WCS keywords
+        """
+        hdu.header['ORIENTAT'] = image_wcs.orientat
+        hdu.header['CD1_1'] = image_wcs.wcs.cd[0][0]
+        hdu.header['CD1_2'] = image_wcs.wcs.cd[0][1]
+        hdu.header['CD2_1'] = image_wcs.wcs.cd[1][0]
+        hdu.header['CD2_2'] = image_wcs.wcs.cd[1][1]
+        hdu.header['CRVAL1'] = image_wcs.wcs.crval[0]
+        hdu.header['CRVAL2'] = image_wcs.wcs.crval[1]
+        hdu.header['CRPIX1'] = image_wcs.wcs.crpix[0]
+        hdu.header['CRPIX2'] = image_wcs.wcs.crpix[1]
+        hdu.header['CTYPE1'] = image_wcs.wcs.ctype[0]
+        hdu.header['CTYPE2'] = image_wcs.wcs.ctype[1]
```

Comment:
@nden Couldn't most of the code here be simplified using `WCS.to_header()`?
```python
from astropy.wcs import WCSHDO_none  # at the top of the file
hdu.header.update(image_wcs.wcs.to_header(WCSHDO_none))
hdu.header['ORIENTAT'] = image_wcs.orientat
hdu.header['VAFACTOR'] = 1.0
```

Reply:
Eh,... ignore my previous comment: `to_header()` gives `PC` coefficients instead of `CD` (they are the same numerically for HST but require extra handling, i.e., renaming).

Reply:
OK - ignored.